dot/sync/chain_processor_integration_test.go

```diff
@@ -33,11 +107,24 @@ func TestChainProcessor_HandleBlockResponse_ValidChain(t *testing.T) {
 	rt, err := responder.blockState.GetRuntime(bestBlockHash)
 	require.NoError(t, err)
 
+	babeCfg, err := rt.BabeConfiguration()
+	require.NoError(t, err)
+
+	timestamp := uint64(time.Now().Unix())
+	slotDuration := babeCfg.SlotDuration
+
 	for i := 0; i < maxResponseSize*2; i++ {
-		block := BuildBlock(t, rt, parent, nil)
+		// calcule the exact slot for each produced block
+		currentSlot := timestamp / slotDuration
+
+		block := buildBlockWithSlotAndTimestamp(t, rt, parent, currentSlot, timestamp)
 		err = responder.blockState.(*state.BlockState).AddBlock(block)
 		require.NoError(t, err)
 		parent = &block.Header
+
+		// increase the timestamp by the slot duration
+		// so we will get a different slot for the next block
+		timestamp += slotDuration
 	}
```

Comment:
Should we move the for loop inner body to a function? It looks repeated a bunch of times in this file